src/ast_calc.h

```diff
@@ -0,0 +1,16 @@
+#ifndef AST_CALC_H
+#define AST_CALC_H
+
+#include "abstract_calc.h"
+#include "config.h"
+
+typedef struct ast_calc_t {
+    abstract_calc_t base;
+    ast_node_t *ast;
+} ast_calc_t;
+
+abstract_calc_t *init_ast_calc (config_t *config);
+int run_ast_calc (abstract_calc_t *calc);
```

Comment:
Those should be inline functions of abstract_calc_t. You can declare them strait in the header:

inline int run_calc (abstract_calc_t * calc) {
  if (NULL == calc)
    return (EXIT_FAILURE);
  return (calc->run (calc));
}